### PR TITLE
[CORE-1494] update kafka client version and scala kafka consumer client

### PR DESCRIPTION
**Pull request by justinmichaud**
See https://github.com/PagerDuty/scala-kafka-consumer/pull/35

### CODE REVIEW

File: scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala [hunk 1]
```diff
@@ -44,6 +47,14 @@ class SchedulerClientSpec extends WordSpecLike with Matchers with ScalaFutures w
...
     override def close(): Unit = ???
     override def close(timeout: Long, unit: TimeUnit): Unit = ???
+    override def sendOffsetsToTransaction(
+        offsets: util.Map[TopicPartition, OffsetAndMetadata],
+        consumerGroupId: String
+      ): Unit = ???
+    override def abortTransaction(): Unit = ???
+    override def initTransactions(): Unit = ???
+    override def beginTransaction(): Unit = ???
+    override def commitTransaction(): Unit = ???
```
Comment (Kev098) [inline scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala]: Does adding all these help pass some test?

File: scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala [hunk 2]
```diff
@@ -65,7 +76,7 @@ class SchedulerClientSpec extends WordSpecLike with Matchers with ScalaFutures w
       val partitionId = task.partitionId(kafkaNumPartitions)
 
       "successful" should {
-        val result = new RecordMetadata(new TopicPartition(kafkaTopic, 0), 0L, 0L)
+        val result = new RecordMetadata(new TopicPartition(kafkaTopic, 0), 0L, 0L, 0, java.lang.Long.valueOf(0L), 0, 0)
```
Comment (Kev098) [inline scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala]: Same question as above

Comment (justinmichaud) [reply scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala]: The new version of the library seemed to have breaking changes

File: scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala (hunk 1 above)
Comment (justinmichaud) [reply scala-api/src/test/scala/com/pagerduty/scheduler/SchedulerClientSpec.scala]: It won't compile otherwise

**Pull request merged**